dashboard/dashboard-front/src/components/GraphArea.vue

```diff
@@ -0,0 +1,178 @@
+<script setup lang="ts">
+import { reactive, watch, computed, ref, onMounted, type Ref } from 'vue'
+import { toastService } from '@/services/ToastService'
+import { apiClient } from '@/services/ApiClient'
+import { DataSet, DataView } from "vis-data"
+import { Network, type Options } from "vis-network"
+
+const graphData: JobGraph = reactive({
+    nodes: [],
+    edges: [],
+})
+const graphContainerRef: Ref<HTMLElement | null> = ref(null)
+const focusedNodes: Ref<string[]> = ref([])
+const legendTitle: Ref<string> = ref('')
+const legendBody: Ref<string> = ref('')
+const physics: Ref<boolean> = ref(true)
+
+interface JobGraph {
+    nodes: JobGraphNode[]
+    edges: JobGraphEdge[]
+}
+
+interface JobGraphNode {
+    id: string
+    type: string
+    title: string
+    subtitle?: string
+}
+
+interface JobGraphEdge {
+    from_id: string
+    to_id: string
+}
+
+function fetchGraph() {
+    apiClient.get(`/api/v1/job/graph`).then(response => {
+        const data: JobGraph = response.data
+        graphData.nodes = data.nodes
+        graphData.edges = data.edges
+    }).catch(err => {
+        toastService.showErrorDetails(`Failed to fetch a jobs graph`, err)
+    })
+}
+
+onMounted(() => {
+    initVisNetwork()
+    fetchGraph()
+})
+
+watch(graphData, () => {
+    initVisNetwork()
+})
+
+watch(physics, () => {
+    networkOptions.physics.enabled = physics.value
+    network?.setOptions(networkOptions)
+})
+
+const nodeStructs = computed(() =>
+    graphData.nodes.map((node: JobGraphNode) => ({
+        id: node.id,
+        label: node.title, 
+        title: node.subtitle, 
+        shape: shapeOfNodeByType(node.type),
+        color: colorOfNodeByType(node.type)
+    }))
+)
+
+const edgeStructs = computed(() =>
+    graphData.edges.map((edge: JobGraphEdge) => ({
+        from: edge.from_id,
+        to: edge.to_id,
+        color: { inherit: "both" },
+        arrows: "to",
+        title: "has access to",
+    }))
+)
+
+var network: Network | null = null
+var networkOptions: Options = {
+    nodes: {
+        borderWidth: 2,
+        shadow: true,
+    },
+    edges: {
+        shadow: true,
+        smooth: {
+            type: "cubicBezier",
+            forceDirection: "none"
+        }
+    },
+    physics: {
+        enabled: physics.value,
+    },
+} as Options
```

Comment:
It's really surprising, but for some reason it is not working without this `as Options`. Probably due  to crappy types coming from upstream lib.
![Screenshot from 2023-05-04 12-20-10](https://user-images.githubusercontent.com/45496492/236177546-eb577ba3-9081-44a9-8f3a-3e9fc94e226c.png)

Reply:
Oh, I see. Then I think you should add the missing parameters. If it's some imported module that is using those parameters it might throw an exception at some point. It looks like you are missing the `enabled` and `roundness` params.